library/src/main/java/com/google/android/exoplayer/text/CuePainter.java

```diff
@@ -350,4 +322,96 @@ private static boolean areCharSequencesEqual(CharSequence first, CharSequence se
     return first == second || (first != null && first.equals(second));
   }
 
+  /**
+   * Calculates left position of the cuebox within enclosing parent layout.
+   *
+   * @param parentWidth Width of parent enclosing the cuebox.
+   * @param position Position value as it comes from the cue object.
+   * @param positionAnchor Position anchor as it comes from the cue object.
+   * @param parentRight Parent right.
+   * @param parentLeft Parent left
+   * @param textWidth Text width
+   * @return Left position of the cue box within parent layout.
+   */
+  private static int calculateCueboxLeftPosition(int parentWidth, float position,
```

Comment:
Hm. The logic in this method seems more confusing than the previous logic. Are you actually fixing anything here? I'm finding it hard to parse what's changed.